ios/Operations/AsyncOperation.swift

```diff
@@ -238,45 +238,22 @@ open class AsyncOperation: Operation {
 
     public let dispatchQueue: DispatchQueue
 
+    private var isReadyObserver: NSKeyValueObservation?
     public init(dispatchQueue: DispatchQueue? = nil) {
         self.dispatchQueue = dispatchQueue ?? DispatchQueue(label: "AsyncOperation.dispatchQueue")
         super.init()
 
-        addObserver(
-            self,
-            forKeyPath: #keyPath(isReady),
-            options: [],
-            context: &Self.observerContext
-        )
-    }
-
-    deinit {
-        removeObserver(self, forKeyPath: #keyPath(isReady), context: &Self.observerContext)
+        isReadyObserver = observe(\.isReady, options: []) { operation, _ in
+            operation.checkReadiness()
+            // Clear the observer when the operation is finished to avoid leaking memory
+            if operation.state == .finished {
```

Comment:
It looks like observation API does not maintain a strong reference to target anymore. Try this in playground:

```swift
class MyOperation: Operation {
    private var observer: NSKeyValueObservation?

    override init() {
        super.init()

        observer = observe(\.isReady, changeHandler: { op, _ in
            print(op)
        })
    }

    deinit {
        print("deinit!")
    }
}

var op: MyOperation? = MyOperation()

op = nil

print("done")
```

Don't forget to remove `observerContext` on the class if we don't need it anymore.